rest/data/host.go

```diff
@@ -165,3 +168,62 @@ func FindHostByIdWithOwner(hostID string, user gimlet.User) (*host.Host, error)
 
 	return hostById, nil
 }
+
+var errHostStatusChangeConflict = errors.New("conflicting host status modification is in progress")
+
+// TerminateSpawnHost enqueues a job to terminate a spawn host.
+func TerminateSpawnHost(ctx context.Context, env evergreen.Environment, u *user.DBUser, h *host.Host) (int, error) {
+	if h.Status == evergreen.HostTerminated {
+		return http.StatusBadRequest, errors.Errorf("host '%s' is already terminated", h.Id)
+	}
+
+	ts := utility.RoundPartOfMinute(1).Format(units.TSFormat)
+	terminateJob := units.NewSpawnHostTerminationJob(h, u.Id, ts)
+	if err := env.RemoteQueue().Put(ctx, terminateJob); err != nil {
+		if amboy.IsDuplicateJobScopeError(err) {
+			err = errHostStatusChangeConflict
+		}
+		return http.StatusInternalServerError, err
+	}
+
+	return http.StatusOK, nil
+}
+
+// StopSpawnHost enqueues a job to stop a running spawn host.
+func StopSpawnHost(ctx context.Context, env evergreen.Environment, u *user.DBUser, h *host.Host) (int, error) {
+	if h.Status == evergreen.HostStopped {
+		return http.StatusBadRequest, errors.Errorf("host '%s' is already stopped", h.Id)
+	}
+	if h.Status != evergreen.HostRunning && h.Status != evergreen.HostStopping {
+		return http.StatusBadRequest, errors.Errorf("host '%s' cannot stop when its status is '%s'", h.Id, h.Status)
+	}
+
+	ts := utility.RoundPartOfMinute(1).Format(units.TSFormat)
+	stopJob := units.NewSpawnhostStopJob(h, u.Id, ts)
+	if err := env.RemoteQueue().Put(ctx, stopJob); err != nil {
+		if amboy.IsDuplicateJobScopeError(err) {
+			err = errHostStatusChangeConflict
+		}
+		return http.StatusInternalServerError, err
+	}
+	return http.StatusOK, nil
+
+}
+
+// StartSpawnHost enqueues a job to start a stopped spawn host.
+func StartSpawnHost(ctx context.Context, env evergreen.Environment, u *user.DBUser, h *host.Host) (int, error) {
+	if h.Status != evergreen.HostStopped {
```

Comment:
do we want to let HostStarting through here too?

Reply:
See my reasoning above why not.